app/assets/javascripts/actions/revisions_actions.js

```diff
@@ -9,9 +11,349 @@ import {
 import { fetchWikidataLabelsForRevisions } from './wikidata_actions';
 import logErrorMessage from '../utils/log_error_message';
 import request from '../utils/request';
+import moment from 'moment';
+import { stringify } from 'query-string';
+import { ORESSupportedWiki, PageAssessmentGrades, PageAssessmentSupportedWiki } from '../utils/article_finder_language_mappings';
+import { url } from '../utils/wiki_utils';
+import { chunk, flatten } from 'lodash-es';
+import { queryUrl } from '../utils/article_finder_utils';
+
+const fetchAll = async (API_URL, params, continue_str) => {
```

Comment:
It might make sense to put all this new functionality into a separate actions file, so that we have one file for interacting with revisions from the Dashboard, and another for getting revisions and related data from MediaWiki.